src/DataSourceProvider.vue

```diff
@@ -13,16 +13,28 @@
 
         <div v-if="dataSources.length || !dataSources.length && !selectedDataSource">
 
-          <Select2
-            :options="dataSources"
-            :selectedOption.sync="selectedDataSource"
-            :customOptionView="formatDataSourceOption"
-            :customSearch="customDataSourceSearch"
-            :optionLabelKey="'name'"
-            :optionValueKey="'id'"
-            :selectWithGroups="!!allDataSources.length"
-          >
-          </Select2>
+          <label for="data-source-select" class="select-proxy-display">
+            <select
+              ref="select"
+              class="hidden-select form-control"
+              @change="onSelectChange"
+              :value="selectedDataSource ? selectedDataSource.id : 'none'"
+            >
+              <option value="none">-- Select data source</option>
```

Comment:
If user doesn't select a data source, it's preferable for the value to be an empty string than "none".

```suggestion
              <option value>-- Select data source</option>
```

Reply:
@tonytlwu Fixed